wagtail/wagtailadmin/widgets.py

```diff
@@ -135,24 +137,48 @@ class AdminPageChooser(AdminChooser):
     choose_another_text = _('Choose another page')
     link_to_chosen_text = _('Edit this page')
 
-    def __init__(self, content_type=None, can_choose_root=False, **kwargs):
+    def __init__(self, target_models=None, content_type=None, can_choose_root=False, **kwargs):
         super(AdminPageChooser, self).__init__(**kwargs)
-        self._content_type = content_type
+
+        self.target_models = list(target_models or [Page])
+
+        if content_type is not None:
+            if target_models is not None:
+                raise ValueError("Can not set both target_models and content_type")
+            warnings.warn(
+                'The content_type argument for AdminPageChooser() is deprecated. Use the target_models argument instead',
+                category=RemovedInWagtail17Warning)
+            if isinstance(content_type, ContentType):
+                self.target_models = [content_type.model_class()]
+            else:
+                self.target_models = [ct.model_class() for ct in content_type]
+
         self.can_choose_root = can_choose_root
 
     @cached_property
     def target_content_types(self):
-        target_content_types = self._content_type or ContentType.objects.get_for_model(Page)
-        # Make sure target_content_types is a list or tuple
-        if not isinstance(target_content_types, (list, tuple)):
-            target_content_types = [target_content_types]
-        return target_content_types
+        warnings.warn(
+            'AdminPageChooser.target_content_types is deprecated. Use AdminPageChooser.target_models instead',
+            category=RemovedInWagtail17Warning)
+        return list(ContentType.objects.get_for_models(*self.target_models).values())
+
+    def _get_lowest_common_page_class(self):
+        """
+        Return a Page class that is an ancestor for all Page classes in
+        ``target_models``, and is also a concrete Page class itself.
+        """
+        if len(self.target_models) == 1:
+            # Shortcut for a single page type
+            return self.target_models[0]
+        else:
+            base = self.target_models[0]
```

Comment:
Fair enough. The concept comes from Graph Theory, and is called the [Lowest Common Ancestor](https://en.wikipedia.org/wiki/Lowest_common_ancestor). Reading the code again, I could probably make it more robust, checking that the ancestor found is actually a Page type, etc. It is not overly important, but it would make things nicer in the case of an extra level MTI, for example:

```
Page
  - StoryPage (is_creatable = False)
    - LongFormStoryPage
    - PictureStoryPage
    - VideoStoryPage
```

By specifying that you want to be able to select `[LongFormStoryPage, PictureStoryPage,VideoStoryPage]` pages in your PageChooserPanel, this would still restrict it to choosing `StoryPage` subclasses, instead of reverting to selecting any `Page` and failing validation later.